[WindowScreenCalibration] Improvements to screen calibration window

CastagnaIT opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->
**Improvements:**
- Reorganized the information on screen
- Now the subtitle bar is a bit smaller and match the text base line to better understand the position
Before:
![base_line](https://user-images.githubusercontent.com/3257156/141495388-356fc95e-1dc9-4957-89f6-5bc0a93b132b.jpg)
Now:
![base_lineNEW](https://user-images.githubusercontent.com/3257156/141495414-a6df7d8b-9fc1-4827-afb3-4330a8f9733d.jpg)
- Updated the image of "pixel ratio ajustment" box in the estouchy skin
- Add new control to allow reset calibration settings from GUI instead use mapped keyboard/remote buttons that usually an user do not know
- On estouchy skin other controls without focus are now slightly visible to better understand where touch

**Bug fixes:**
- Fix wrong controls positions when switching from windowed to fullscreen (or reverse order)
- Fix missing scaling of controls images, caused of too bigger controls with a small window or too small controls with higher resolutions
- Fix subtitle bar not matching the subtitle text base line
- Fix subtitle bar can not reach the bottom edge of the screen
- Fix subtitle bar that can exceed the top edge of the screen
- Fix impossibility to reset calibration settings (subtile bar, pixel ratio) when Kodi is in fullscreen mode
- Fix impossibility to move the subtitle bar because was under the central "box" image
- Fix cropped label's text rendering due to missing re-scaling of fonts while changing overscan values

~**Unresolved problems:**~
- ~Resizing overscan cause weird text cuttering, this is a very old Kodi bug, seem that if the text change the text will be rendered correctly but i am not able to found the real cause~

Sidenote:
With the new reset calibration "button" image, i have used a fixed `mover` control instead of a `button` control, this because the button control have weird strange effects with focus/selection in this particular window, maybe caused by the customizations like the scaling management.

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Another thing related to subtitles to improve,
then i take opportunity to improve a bit the window things

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
By using window, i have not tested with touchscreen, but following the logic of transparencies to allow bigger hit rect size should works

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->
a better experience

## Screenshots (if appropriate):
Estuary Windowed (small window size) BEFORE:
![immagine](https://user-images.githubusercontent.com/3257156/141496991-481aa64d-5e33-49b6-9424-8a69237b9f2d.png)
Estuary Windowed (small window size) AFTER:
![immagine](https://user-images.githubusercontent.com/3257156/144245940-46425c92-dbd4-4ce7-a7f0-7982dc47f4d1.png)
Estuary Fullscreen:
![immagine](https://user-images.githubusercontent.com/3257156/144246009-1f9ee024-5221-41b1-acd2-a54d1c1e92fa.png)
Estouchy Fullscreen:
![immagine](https://user-images.githubusercontent.com/3257156/144241964-c2b7006f-eac3-4e29-9249-fdb280909500.png)

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

CastagnaIT committed:
[WindowScreenCalibration] Fix not reachable code

CastagnaIT committed:
[WindowScreenCalibration] Reset controls after window resize

CastagnaIT committed:
[WindowScreenCalibration] Updated estouchy pixel aspect ratio image

CastagnaIT committed:
[WindowScreenCalibration] Allow reset subtitle/pixel ratio with all resolutions

enen92 commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -13028,12 +13029,10 @@ msgid "Calibration reset"
 
 msgctxt "#20326"
-msgid "This will reset the calibration values for {0:s}"
+msgid "Do you want reset the calibration values to its default values for \"{0:s}\" resolution?"
```
```suggestion
msgid "Do you want reset the calibration values to their default values for \"{0:s}\" resolution?"
```

enen92 replied:
or simply to their defaults for xxx resolution

enen92 commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -22852,3 +22851,10 @@ msgstr ""
 msgstr ""
+
+#. Window screen calibration
+#: addons/skin.estuary/xml/SettingsScreenCalibration.xml
```
I don't think this is correct. Addon/skin specific localized values should be handled by the addon translations, not core.

enen92 replied:
Or at least, if this is a core string, the xml in the skin should not be listed here as it is not part of core

CastagnaIT replied:
this made me weird too, so i waited for feedback from you,
all ID's of "calibration window" are here on core, i will move this new ID to addon skin localization,
since i'm going to have to do a little code cleanup PR i could move all others ID's to addon skin localization

enen92 commented on xbmc/settings/windows/GUIWindowSettingsScreenCalibration.cpp:
```diff
@@ -178,15 +203,18 @@ bool CGUIWindowSettingsScreenCalibration::OnMessage(CGUIMessage& message)
-      if (message.GetParam1() == GUI_MSG_WINDOW_RESIZE)
+      if (message.GetParam1() == GUI_MSG_WINDOW_RESIZE && IsActive())
       {
+        m_Res.clear();
```
maybe group this into some private method just like you have above for `NextControl()`

enen92 commented on xbmc/settings/windows/GUIWindowSettingsScreenCalibration.cpp:
```diff
@@ -245,9 +283,38 @@ void CGUIWindowSettingsScreenCalibration::EnableControl(int iControl)
   SET_CONTROL_FOCUS(iControl, 0);
 }
 
+void CGUIWindowSettingsScreenCalibration::StoreControlsSize()
```
can you explain the need for storing control sizes? I mean the fact this is needed sounds suspicious/hacky. Maybe what is needed is the persistence of some other values those controls refer to, not really the controls themselves

enen92 replied:
Ok, I see this doesn't really store anything anywhere, just memory. I think this can be private as it is specific to this window implementation. Please also check if you don't need some sort of locking on the map (probably the increased read consistency - you only write the values here - won't justify the impact on performance and GUI locks are bad anyway.

enen92 reviewed (changes requested):
Some initial review comments

CastagnaIT commented:
@enen92 i have updated description (see unresolved problems) and screenshots
there is a problem with the Estouchy skin its resolution wrongly influences the scaling of controls
besides what i have written in the description i don't know other solutions at the moment
if you have some suggestions write on slack

bossanova808 commented:
I don't want to bike shed here, but as a user I've never understood why this screen uses a line, rather than actual sample text (lorum ipsem, say), - to actually show where the subtitles will be.  To me, it's never clear to the user - is the line the centre of the subs?  The bottom (as it seems to be from the above)?  The top?  

And if two lines, or three are present, what happens there?  (And to which subs alignment mode does this relate is not super clear here, either).

CastagnaIT commented:
> but as a user I've never understood why this screen uses a line, rather than actual sample text (lorum ipsem, say),

because until to Kodi 19 the subtitle rendering was done via an hacked way of GUI rendering, so it was difficult to add a sample text to render correctly. now we have better rendering, i wouldn't know if it's easier to implement sample text rendering, but anyway it is not in the scope of this PR

in any case no one prevents users from performing the calibration while a video is playing,
just enable subtitles and pause the video, to have a persistent subtitle while you are calibrating

>  To me, it's never clear to the user - is the line the centre of the subs? The bottom (as it seems to be from the above)? The top?

the bar should always match the bottom base of the text line, like the screenshot shown at the beginning, in previous Kodi it was badly handled

> And if two lines, or three are present, what happens there? (And to which subs alignment mode does this relate is not super clear here, either).

even if there were several lines of text, the bar always represents the lowest line text

CastagnaIT commented:
@enen92 i have solved the remaining problems, now PR is ready for the review

enen92 commented on addons/skin.estouchy/language/resource.language.en_gb/strings.po:
```diff
@@ -386,7 +386,51 @@ msgctxt "#31565"
+#. Window screen calibration: calibration from the rectangle box
+#: xml/SettingsScreenCalibration.xml
+msgctxt "#31745"
+msgid "[B]Pixel ratio adjustment[/B][CR]Adjust the rectangle so it is perfectly square"
```
I don't think it's a good practice to mix styling with translations. Although in addons that are 100% under your control this might be fine, in generic code I'm tempted to say we should avoid that.
Why not use the bold and carriage return tags within the skin xml files itself instead of forcing them right away on the gettext files?

CastagnaIT replied:
the text is set by the source code not in the xml, but i think i can split all sentences in to two ID's, then formatting it from code

CastagnaIT replied:
removed style in the translations

enen92 commented on xbmc/guilib/GUITextLayout.h:
```diff
@@ -132,14 +132,15 @@ class CGUITextLayout
                        uint32_t align);
   static void Filter(std::string &text);
 
+  void CalcTextExtent();
```
Anytime something like this happens (moving some private member to public access we should question if this is really necessary or if we can't do this with a better approach. For an API consumer, "CalcTextExtend" doesn't ring a bell so why should it be available to the outside? Can't a better approach, using messaging be used instead?
I'll have to take a deeper look into this though as I am not familiar with this part of the code

CastagnaIT replied:
i can try to add the `OnMessage` callback but my concern is, how can you understand if the callback arrives before or after the same callback on GUIFontManager? What order logic is used to raise the callbacks of the messages sent?

enen92 replied:
Don't get me wrong, I am not saying it's the way to go specially because I have no idea what I am talking about :)
I am basically saying this method is really low level and we should get to execute it some other way. Being it messaging, encapsulation, etc - basically as the consequence of some event. It's not something that you'd want to see exposed on a public api - furthermore it even retuns void as a response to "calc" something. Just seems too weird.

Corrently this method is called internally in `UpdateStyled`. Can't you simply call `UpdateStyled` on your side so that the method is internally triggered? Or wrap it into some other meaningfull method like `Refresh()` ?

CastagnaIT replied:
`UpdateStyled` do things that are not needed because is indeed to be used when the text change only,
here lack a way to re-update the internal cached text size when the screen scaling change (thing done by `CalcTextExtent`). 

Currently the GUITextLayout can update the chached text extent (sizes used for rendering) only when the text change and this is not so appropriate, when the screen font scaling change is kept the old values because the text is always the same, then will cause the cropped text rendering.

The `RefreshLayoutSize()` call https://github.com/xbmc/xbmc/pull/20509/files#diff-34de2075d888d6cf162eeb2df31a4901fe3636f0cbdc3acdde321e1b99134abaR78
update the text extent used to determinate the text size,
then this influence behaviour here:
https://github.com/xbmc/xbmc/pull/20509/files#diff-34de2075d888d6cf162eeb2df31a4901fe3636f0cbdc3acdde321e1b99134abaL100-L110
the calls `GetMaxWidth()` recall GUITextLayout to get the text size (now updated by RefreshLayoutSize),
and then `m_label.SetMaxRect()` call inside it, will call `UpdateRenderRect` used to update the rendering rect
also in this case pick the text size from GUITextLayout

maybe i can try use the Messaging system by addin a new type of message, then use GUIFontManager to ask to update all GUITextLayout internal size, i will have to re-check the whole chain of events because is a mess of intricate things...

CastagnaIT replied:
i have tried an alternative solution by using messagins at lower level so to CGUITextLayout e.g. https://github.com/CastagnaIT/xbmc/commit/5feeec4ad283008b317d2efde4eb9960d070340d
(that i find it ugly because we have to add/remove it on each control)
but this not works,
because when we update the value on `CGUITextLayout` by use OnMessage (with `CalcTextExtent`) the `m_maxRect` in `CGUILabel` will be not updated, and  `m_maxRect` value is updated from `CGUILabelControl` with `SetMaxRect` due to the custom value behaviour

if we theorise about run CalcTextExtent from `CGUILabel` and then make some kind of adaptations to update `m_maxRect` in any case `CalcTextExtent` method will be public, and i would not know how to obtain the value CGUILabelControl::GetMaxWidth from `CGUILabel`
I don't think the situation will improve much

CastagnaIT replied:
i have found another way, by taking inspiration from `CGUITextBox` class i have derived the control from `CGUITextLayout` then in this way i can keep `CalcTextExtent` protected as in origin.
i think now should be good see last commit added

enen92 commented on xbmc/guilib/GUILabelControl.cpp:
```diff
@@ -203,7 +210,15 @@ bool CGUILabelControl::OnMessage(CGUIMessage& message)
+  {
+    // If the gui font size/aspect has been rescaled (e.g. screen calibration
+    // window) we have to refresh the text layout sizes
+    m_refreshLayout = true;
```
Is this needed? Isn't it possible to simply call updateinfo and force refresh the layout there? What is the reason for the controlling bool? Is updateinfocalled on a loop?

CastagnaIT replied:
already tried but not works, seem this is like a cat and mice
from what i understand seem that kodi call from CApplication:
as first CGUIWindowSettingsScreenCalibration::DoProcess()
as second CGUIWindowManager::FrameMove()

when GUI_MSG_WINDOW_RESIZE is called by CGUIWindowSettingsScreenCalibration::FrameMove
if m_refreshLayout is not stored when CGUILabelControl::UpdateInfo is called (from DoProcess() ) the layout refresh seem not happen correclty

i have tried also move GUI_MSG_WINDOW_RESIZE call from CGUIWindowSettingsScreenCalibration::FrameMove to CGUIWindowSettingsScreenCalibration::DoProcess but without success

i have done a little scheme of the calls sequences to try understand better:
```
(1st call) CApplication call -> CGUIWindowSettingsScreenCalibration::DoProcess -> CGUIWindow::DoProcess -> CGUIControl::DoProcess -> CGUIControlGroup::Process
then CGUIControlGroup::Process execute for each control:
1st call - UpdateVisibility -> CGUIControl::UpdateVisibility -> UpdateInfo -> CGUILabelControl::UpdateInfo  (bInvalidated is true then RefreshLayoutSize)
2nd call - DoProcess -> CGUILabelControl::Process ->then  CGUIControl::DoProcess  update m_renderRegion (render region) ?? then bInvalidated will be false
3rd call - CGUIControl::Process

(2nd call) CApplication call -> FrameMove -> CGUIWindowManager::FrameMove() -> CGUIWindowSettingsScreenCalibration::FrameMove()
```

I have no other ideas at the moment

CastagnaIT commented on xbmc/guilib/GUIWindow.cpp:
```diff
@@ -618,6 +618,15 @@ bool CGUIWindow::OnMessage(CGUIMessage& message)
       if (HasID(message.GetSenderId()))
       {
         m_focusedControl = message.GetControlId();
+        // We ensure that all others childrens have focus disabled,
```
related to this, the use case is simple:
move the subtitle bar inside/over the kodi logo box (ratio control),
then move the mouse pointer -before- over the logo then -after- over the subtitle bar
will result that the ratio control and the subtitle bar have both the focus

enen92 replied:
I ask this because I've found this problem earlier in other contexts, eg when you have two button too close to each other, kodi will focus both. This is easily reproduced in those dialogs where you have the "ok" and "cancel" buttons on the right of the dialog.

![image](https://user-images.githubusercontent.com/7375276/148360379-1358e3c8-e973-4168-b8af-53a4f307bd7c.png)

CastagnaIT replied:
I didn't notice that, but i had seen this:
![immagine](https://user-images.githubusercontent.com/3257156/148361082-e954168b-e226-4db4-94cc-a26c1fbd3f1f.png)
i think should same problem, but this change not fix these cases, maybe because are within a list control?

enen92 commented on xbmc/settings/windows/GUIWindowSettingsScreenCalibration.cpp:
```diff
@@ -178,15 +217,19 @@ bool CGUIWindowSettingsScreenCalibration::OnMessage(CGUIMessage& message)
+          message.GetSenderId() != WINDOW_SCREEN_CALIBRATION && IsActive())
       {
+        m_Res.clear();
+        CServiceBroker::GetWinSystem()->GetGfxContext().GetAllowedResolutions(m_Res);
```
This is something we should probably try to get rid of later. It's not something you have introduced with this changes (so this is not a change request nor will it block the PR).

It does not make much sense to be calling upwards to get the context or ask for allowed/or current resolutions. Ideally this should be propagated to this component somehow. Dialogs/windows need to react to resolution changes in some event driven way (informed by the parent)

enen92 commented:
@CastagnaIT I've taken a look, please check if this solves your issue. Seems to me like the proper way to do this (and avoiding calling "private" methods):

```
diff --git a/xbmc/guilib/GUILabel.cpp b/xbmc/guilib/GUILabel.cpp
index f8f3ab7094..c8c733c3db 100644
--- a/xbmc/guilib/GUILabel.cpp
+++ b/xbmc/guilib/GUILabel.cpp
@@ -148,11 +148,6 @@ bool CGUILabel::UpdateColors()
   return m_label.UpdateColors();
 }
 
-void CGUILabel::RefreshLayoutSize()
-{
-  CGUITextLayout::CalcTextExtent();
-}
-
 bool CGUILabel::SetMaxRect(float x, float y, float w, float h)
 {
   CRect oldRect = m_maxRect;
diff --git a/xbmc/guilib/GUILabel.h b/xbmc/guilib/GUILabel.h
index 0f4f13d3f6..a5eac98e0a 100644
--- a/xbmc/guilib/GUILabel.h
+++ b/xbmc/guilib/GUILabel.h
@@ -215,10 +215,6 @@ public:
    */
   static bool CheckAndCorrectOverlap(CGUILabel &label1, CGUILabel &label2);
 
-  /*! \brief Recalculate the text layout size
-   */
-  void RefreshLayoutSize();
-
 protected:
   UTILS::COLOR::Color GetColor() const;
 
diff --git a/xbmc/guilib/GUILabelControl.cpp b/xbmc/guilib/GUILabelControl.cpp
index cf3ba75cb5..1f84829094 100644
--- a/xbmc/guilib/GUILabelControl.cpp
+++ b/xbmc/guilib/GUILabelControl.cpp
@@ -70,12 +70,6 @@ void CGUILabelControl::UpdateInfo(const CGUIListItem *item)
 
   bool changed = false;
 
-  if (m_refreshLayout)
-  {
-    m_label.RefreshLayoutSize();
-    m_refreshLayout = false;
-  }
-
   if (m_startHighlight < m_endHighlight || m_startSelection < m_endSelection || m_bShowCursor)
   {
     std::wstring utf16;
@@ -114,10 +108,16 @@ void CGUILabelControl::UpdateInfo(const CGUIListItem *item)
       label = ShortenPath(label);
 
     changed |= m_label.SetMaxRect(m_posX, m_posY, GetMaxWidth(), m_height);
-    changed |= m_label.SetText(label);
   }
+
+  // try to update text as the control migth be invalid
+  // note: it will return early anyway if no changes and not invalid
+  changed |= m_label.SetText(label);
+
   if (changed)
+  {
     MarkDirtyRegion();
+  }
 }
 
 void CGUILabelControl::Process(unsigned int currentTime, CDirtyRegionList &dirtyregions)
@@ -215,9 +215,7 @@ bool CGUILabelControl::OnMessage(CGUIMessage& message)
   }
   else if (message.GetMessage() == GUI_MSG_WINDOW_RESIZE && IsVisible())
   {
-    // If the gui font size/aspect has been rescaled (e.g. screen calibration
-    // window) we have to refresh the text layout sizes
-    m_refreshLayout = true;
+    m_label.SetInvalid();
   }
 
   return CGUIControl::OnMessage(message);
diff --git a/xbmc/guilib/GUILabelControl.h b/xbmc/guilib/GUILabelControl.h
index 2acefa0102..ce7ba4ec87 100644
--- a/xbmc/guilib/GUILabelControl.h
+++ b/xbmc/guilib/GUILabelControl.h
@@ -76,6 +76,5 @@ protected:
   unsigned int m_endHighlight;
   unsigned int m_startSelection;
   unsigned int m_endSelection;
-  bool m_refreshLayout{false};
 };
 ```
Also check if you still need to derive from textlayout (if not it'd make the diff on this PR far smaller)

CastagnaIT committed:
[WindowScreenCalibration] Fixed cropped label text rendering

due to missing re-scaling of fonts while changing overscan values
and missing update of CGUILabelControl internal rendering size values

CastagnaIT commented:
@enen92 thanks for the snippet yes it is ok
i have applied and removed all other changes, the only difference compared to your snipped
is that i have keep unchanged the line `changed |= m_label.SetText(label);` as original position, 
because SetStyledText in any case call CalcTextExtent, and also because otherwise this cause text flickering on text input controls like e.g. ip address in numeric input window, due to double text changing

enen92 reviewed (approved):
Looks good to me. Would be nice if someone can take a look at the skin changes, requested review from ronie.

ronie commented:
a few comments:

1) i don't think it's a good idea to remove those strings from our default language file.
each and every skin needs those strings, so they are not skin specific and thus should not be moved to the skin language file

2) i can't seem to focus the new reset button? tried both Estuary and Estouchy using a keyboard and mouse.

3) the reset button is oddly placed in Estouchy on a 4:3 screen
![SC](https://user-images.githubusercontent.com/687265/148417928-a03e9d3c-2d44-4f43-9c8c-66f163d53a6e.jpg)

CastagnaIT commented:
for point 1: @enen92 initially have suggest to move id's in to skins, it is not a problem for me to restore to common language file, but there seems to be different opinions..

for point 2: your screenshot look strange, seems to be missing something that corrupts the functionality of the source code, suggest you to rebuild kodi and copy all new files

for point 3: could be the same problem as above, but I think I'm able to do a test on an old 4:3 monitor

enen92 commented:
> for point 1: @enen92 initially have suggest to move id's in to skins, it is not a problem for me to restore to common language file, but there seems to be different opinions..
> 
ronie is far more experienced and knowledgeable in kodi skinning than myself, so please follow his advice. Apologies for the double change.

I can't reproduce the points raised though.

ronie commented:
> for point 3: could be the same problem as above, but I think I'm able to do a test on an old 4:3 monitor

Estouchy is coded for these resolutions:
https://github.com/xbmc/xbmc/blob/a795c1aadc96f6d836d2d109b8acc2b9ae28a776/addons/skin.estouchy/addon.xml#L7-L8

so if you place the reset button with a y=900px and a height of 100px, it will be partly off screen.
may i suggest to change it to `<posy>850</posy>`

to test those specific resolutions, just place this in advancedsettings.xml
```
	<window>
		<width>1280</width>
		<height>960</height>
	</window>
```
and run kodi in windowed mode.
much easier than digging up an old 4:3 CRT :-)

ronie commented:
> for point 2: your screenshot look strange

ah yeah, my mistake, forgot to refresh the window after switching between fullscreen and windowed mode.
but it's unrelated to the issue...

here's some proper screenshots:

works ok in fullscreen window:
![SC4](https://user-images.githubusercontent.com/687265/148427131-b83a6a02-7cd6-44f6-951b-eec8608992c2.jpg)

does not work in windowed mode:
![SC5](https://user-images.githubusercontent.com/687265/148427173-1a4a220f-72aa-4a73-b2bf-0453966776f0.jpg)

CastagnaIT commented:
@ronie thanks for resolution tip! in any case i have tested also on a real 4:3 monitor but i still cannot reproduce your problems

> so if you place the reset button with a y=900px and a height of 100px, it will be partly off screen.

the reset button positions x/y are dinamically set from the source code, then skin x/y values are ignored and overwritten when the window is opened,
but from your screenshots the text still to be different from files of this PR,
seem that in your build Kodi still have mixed with old files, at least what concern language file:
![immagine](https://user-images.githubusercontent.com/3257156/148646943-08a3e0ed-1bbe-4c94-9a0e-bb3964fc707b.png)
this could affect the right execution of the source code and then cause this problems

CastagnaIT committed:
[WindowScreenCalibration] Improvements to screen calibration window

CastagnaIT commented:
language changes moved to the common language file
@ronie when you will have time please test again

ronie commented:
@CastagnaIT i've recompiled and everything works as expected now, thx!
